src/order/closure.lean

```diff
@@ -118,6 +118,28 @@ c.closure_top
 lemma closure_le_closed_iff_le {x y : α} (hy : c.closed y) : x ≤ y ↔ c x ≤ y :=
 by rw [← c.closure_eq_self_of_mem_closed hy, le_closure_iff]
 
+lemma closure_closure_union_left {α : Type u} [semilattice_sup α] (c : closure_operator α)
+  (x y : α) :
+  c (c x ⊔ y) = c (x ⊔ y) :=
+le_antisymm ((closure_le_closed_iff_le c (closure_is_closed _ _)).1 (sup_le (c.monotone le_sup_left)
+  (le_trans le_sup_right (le_closure _ _)))) (c.monotone (sup_le_sup_right (le_closure _ _) _))
+
+lemma closure_closure_union_right {α : Type u} [semilattice_sup α] (c : closure_operator α)
+  (x y : α) :
+  c (x ⊔ c y) = c (x ⊔ y) :=
+by {rw [sup_comm, closure_closure_union_left, sup_comm]}
+
+lemma closure_closure_union {α : Type u} [semilattice_sup α] (c : closure_operator α)
+  (x y : α) :
+  c (c x ⊔ c y) = c (x ⊔ y) :=
+by {rw [closure_closure_union_left, closure_closure_union_right]}
+
+lemma closure_closure_bUnion {α : Type u} [complete_lattice α] (c : closure_operator α)
+  (s : set α) :
+  c (⨆ x ∈ s, c x) = c (⨆ x ∈ s, x) :=
```

Comment:
Doesn't this hold more generally for arbitrary supr?

Reply:
That is,
```suggestion
  c (⨆ i, c (x i)) = c (⨆ i, x i) :=
```
For some indexed family x

Reply:
It indeed works!